adapters/core2sn/class.go

```diff
@@ -2,12 +2,78 @@ package core2sn
 
 import (
 	"github.com/NethermindEth/juno/core"
+	"github.com/NethermindEth/juno/core/felt"
 	"github.com/NethermindEth/juno/starknet"
+	"github.com/NethermindEth/juno/utils"
 )
 
-func AdaptSierraEntryPoint(ep core.SierraEntryPoint) starknet.SierraEntryPoint {
-	return starknet.SierraEntryPoint{
-		Index:    ep.Index,
-		Selector: ep.Selector,
+func AdaptCompiledClass(coreCompiledClass *core.CompiledClass) starknet.CompiledClass {
+	feederCompiledClass := new(starknet.CompiledClass)
```

Comment:
Let's not heap allocate this if we are going to return a value from this function by de-referencing it.

Reply:
I don't this is necessarily true. IIUC The reference doesn't escape the function so it won't be allocated on the heap

Reply:
The number of allocations will be the same (I benchmarked it with `b.ReportAllocs()`. However, in C++ there is optimization called RVO (return value optimization) and NRVO (named return value optimization) which essentially constructs an object in place of caller (without copying). I don't know whether the Go compiler uses it, but just in case I rewrote this function.